src/main/scala/nodes/learning/KernelGenerator.scala

```diff
@@ -0,0 +1,273 @@
+package nodes.learning
+
+import breeze.linalg._
+import breeze.math._
+import breeze.numerics._
+import org.apache.spark.rdd.RDD
+import org.apache.spark.SparkConf
+import org.apache.spark.SparkContext
+import org.apache.spark.SparkContext._
+import utils._
+
+
+// Trait to be implemented by Kernel functions
+trait KernelGenerator[T] {
```

Comment:
i think my mental model of a kernel isn't really a transformation, since the whole point is that we can't explicitly store the transformation. And its not really analogous to any of the other transformers we have (SIFT/CKN/RandomCosine), idk it seems just a bit contrived. But I can do it.